public/app/plugins/datasource/cloudwatch/components/LogGroupSelector.tsx

```diff
@@ -0,0 +1,156 @@
+import { debounce, intersection, unionBy } from 'lodash';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { SelectableValue, toOption } from '@grafana/data';
+import { MultiSelect } from '@grafana/ui';
+import { InputActionMeta } from '@grafana/ui/src/components/Select/types';
+import { notifyApp } from 'app/core/actions';
+import { createErrorNotification } from 'app/core/copy/appNotification';
+import { dispatch } from 'app/store/store';
+
+import { CloudWatchDatasource } from '../datasource';
+import { appendTemplateVariables } from '../utils/utils';
+
+const MAX_LOG_GROUPS = 20;
+const MAX_VISIBLE_LOG_GROUPS = 4;
+const DEBOUNCE_TIMER = 300;
+
+export interface LogGroupSelectorProps {
+  region: string;
+  selectedLogGroups: string[];
+  onChange: (logGroups: string[]) => void;
+
+  datasource?: CloudWatchDatasource;
+  allowCustomValue?: boolean;
```

Comment:
Thinks this should be enabled in both LogsQueryField and in the ConfigEditor so let's remove this prop?